jujugui/static/gui/src/app/jujulib/bakery-factory.js

```diff
@@ -0,0 +1,60 @@
+/* Copyright (C) 2016 Canonical Ltd. */
+'use strict';
+
+var module = module;
+
+(function (exports) {
+
+
+  var jujulib = exports.jujulib;
+  /**
+   * BakeryFactory to create and maintain bakery objects.
+   *
+   */
+
+  /**
+   Initialize.
+
+   @method initializer
+   @param {Object} cfg a config object containing the Bakery module
+                   {Bakery: Y.juju.environments.web.Bakery}
+   @return {undefined} Nothing.
+   */
+  function bakeryFactory(cfg) {
+    this.Bakery = cfg.Bakery;
+    this.bakeries = new Map();
+  };
+
+  bakeryFactory.prototype = {
+    /**
+     Return a bakery instance associated with the cfg.serviceName provided.
+     @method get
+     @param cfg the config needed to create a new bakery instance.
+     @return {Object} the bakery object associated with the cfg.serviceName i
+     nstance, created if needed.
+     */
+    get: function (cfg) {
+      return this.bakeries.get(cfg.serviceName);
+    },
+
+    create: function (cfg) {
```

Comment:
Missing docstring, please describe what are the supported fields in cfg, what are optional, etc.